### PR TITLE
Refactored promise chains (.then) replaced with async/await

### PR DESCRIPTION
**Pull request by alexanderpalsson**

### CODE REVIEW

File: src/hello-visualization/app.js [hunk 1]
```diff
@@ -170,51 +169,52 @@ angular.module('app', []).component('app', {
...
-                .then(() => {
-                  result.createSessionObject(scatterplotProperties).then((model) => {
-                    scatterplotObject = model;
+      (async () => {
+        const data = await $http.get('https://gist.githubusercontent.com/carlioth/b86ede12e75b5756c9f34c0d65a22bb3/raw/e733b74c7c1c5494669b36893a31de5427b7b4fc/MovieInfo.csv');
+        const table = new Halyard.Table(data.data, { name: 'MoviesInfo', delimiter: ';', characterSet: 'utf8' });
+        halyard.addTable(table);
+        let qix;
+        try {
+          qix = await enigma.create(config).open(); // lägg till try t
```
Comment (wennmo) [inline src/hello-visualization/app.js]: Remove comment

File: src/hello-visualization/app.js [hunk 2]
```diff
@@ -36,15 +38,12 @@ angular.module('app', []).component('app', {
...
-          this.getMovieInfo().then(() => {
-            $scope.showFooter = true;
-          });
-        });
-      });
+    const select = async (value) => {
+      const field = await app.getField('Movie');
+      await field.select(value);
+      $scope.dataSelected = true;
+      $scope.showFooter = true;
```
[the remark this answers is not in the document]
Comment (hrigner) [reply src/hello-visualization/app.js]: It does.. if getMovieInfo is slow the footer will show without any data. I would suggest to make getMovieInfo return the layout. Await the layout in select, and then do the Scatterplot.showDetails and showFooter. Sth like 
```
const movieInfo = await this.getMovieInfo();
Scatterplot.ShowDetails(movieInfo);
$scope.showFooter = true;
```

File: src/hello-visualization/app.js [hunk 3]
```diff
@@ -170,51 +169,52 @@ angular.module('app', []).component('app', {
...
+        } catch (error) {
+          this.error = 'Could not connect to QIX Engine';
+          this.connecting = false;
+        }
+        // läg till try cath
 
-                    const update = () => scatterplotObject.getLayout().then((layout) => {
-                      paintScatterPlot(layout);
-                    });
+        let result;
```
Comment (hrigner) [inline src/hello-visualization/app.js]: You can skip `result` and use app directly. So, 
`app = await qix.createSessionAppUsingHalyard(halyard);`

File: src/hello-visualization/app.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexanderpalsson) [reply src/hello-visualization/app.js]: If showfooter is called after getMovieInfo it doesn't work as intended. I think it because

**Pull request merged**